### PR TITLE
Ignore hashtag on pages

### PR DESCRIPTION
**Pull request by ikreymer**
This PR allows matching of pages in a provided pages.jsonl to WARC records, ignoring the hashtag, eg. `https://example.com/#hashtag` is matched to WARC record `https://example.com/`. The page is added as `https://example.com/#hashtag` but the timestamp of the record without the hashtag is used.

bumps version 0.4.9

### CODE REVIEW

File: wacz/waczindexer.py [hunk 1]
```diff
@@ -205,12 +205,13 @@ def check_pages_and_text(self, record):
         matched_id = check_http_and_https(url, ts, self.passed_pages_dict)
         # If we find a match build a record
         if matched_id:
-            new_page = {"timestamp": ts, "url": url, "title": url}
             input_page = self.passed_pages_dict[matched_id]
+            new_page = {
+                "timestamp": ts,
+                "url": input_page.get("url") or url,
+                "title": input_page.get("title") or url,
```
Comment (tw4l) [inline wacz/waczindexer.py:212]: Curious what's the motivation to falling back to URL for the title if a title's not present, since we didn't before?

**Review by tw4l — APPROVED** [review]
Code looks good! One comment, and might benefit from a test?

File: wacz/waczindexer.py (hunk 1 above)
Comment (ikreymer) [reply wacz/waczindexer.py:212]: It is actually doing that before, just a different way, first setting title as url and then overriding it. But agree that we should add tests for this..

**Comment by ikreymer** [comment]
Added some tests, but also additional refactoring to be able to include additional fields in pages.jsonl
Previously, all but the core fields were being filtered out during WACZ creation process..

**Pull request merged**